### DISCUSSION TITLE
Durable Objects

### DISCUSSION THREAD

**Discussion by sdas41395** [flareact/flareact#243] [Q&A]
Hi I was wondering if flareact had any support for durable objects?

**Comment by traffisco** [comment]
How would you like to use them in the context of flareact? You should be able to use full cloudflare workers capabilities whenever the code run "server side" so if you need to fetch an object, its should be available

**Reply by sdas41395** [reply]
I'm trying to utilize them in this context here: https://github.com/cloudflare/durable-objects-webpack-commonjs
Essentially we have a flareact project that resides on a worker that we also want to include a durable object websocket on. This way we can not only serve the project but keep an open connection on that specific worker when we need to refresh the page. 

Im running into the issue where durable objects require a main module specified in order to use them in our worker. Because our worker is running a flareact application we can't seem to combine the required *.mjs code with our flareact project and I'm beginning to suspect this might not be possible. 

Is this something that is doable / has already been done?

**Reply by traffisco** [reply]
You might get away with using the "api" folder or creating another non flareact worker and assigning it to a specific route on the domain.

**Reply by sdas41395** [reply]
Ah I see. I'll give those a shot. Thank you!
